### PR TITLE
add cpp build system with bazel

### PR DESCRIPTION
**Pull request by QiJune**

### CODE REVIEW

File: elasticdl/cc/common/tensor.h [hunk 1]
```diff
@@ -0,0 +1,72 @@
+#pragma once
```
Comment (skydoorkai) [inline elasticdl/cc/common/tensor.h]: Should we use `gragma once` or include guards? Seems that the only difference is that if the header file is copied around different locations by the compiler, `gragma once` won't work.

Comment (QiJune) [reply elasticdl/cc/common/tensor.h]: I have no preference for using `#pragma once` or include guards. If google style suggests include guards, I will follow it.

File: elasticdl/cc/common/type.cc [hunk 2]
```diff
@@ -0,0 +1 @@
+#include "elasticdl/cc/common/type.h"
```
Comment (skydoorkai) [inline elasticdl/cc/common/type.cc]: Why need `type.cc` which contains only one line inlude ?

Comment (QiJune) [reply elasticdl/cc/common/type.cc]: Fix it.

**Comment by QiJune** [comment]
Close this PR. We plan to make C++ very thin, and Makefile is enough.

**Pull request closed without merging**